sonic-xcvrd/xcvrd/xcvrd.py

```diff
@@ -1758,6 +1761,64 @@ def join(self):
         if self.exc:
             raise self.exc
 
+    def check_transceiver_temperature(self, logical_port_name, th_table, dom_info_cache, temperature_status):
+        TEMP_NORMAL = 0
+        TEMP_HIGH_ALARM = 1
+        TEMP_LOW_ALARM = 2
+        TEMP_HIGH_WARNING = 3
+        TEMP_LOW_WARNING = 4
+
+        TEMP_ERROR_TO_DESCRIPTION_DICT = {
+            TEMP_NORMAL: "temperature normal",
+            TEMP_HIGH_ALARM: "temperature high alarm",
+            TEMP_LOW_ALARM: "temperature low alarm",
+            TEMP_HIGH_WARNING: "temperature high warning",
+            TEMP_LOW_WARNING: "temperature low warning"
+        }
+
+        for physical_port, physical_port_name in get_physical_port_name_dict(logical_port_name, self.port_mapping).items():
+            orig_temp_status = temperature_status.get(physical_port)
+            if orig_temp_status is None:
+                orig_temp_status = TEMP_NORMAL
+                temperature_status[physical_port] = orig_temp_status
+            new_temp_status = TEMP_NORMAL
+
+            dom_info_dict = dom_info_cache.get(physical_port)
+            presence, threshold = th_table.get(physical_port_name)
+            if presence:
+                dom_th_info_dict = dict(threshold)
+            else:
+                dom_th_info_dict = None
+            if dom_info_dict is not None and dom_th_info_dict is not None:
+                temperature = dom_info_dict.get("temperature")
+                temphighalarm = dom_th_info_dict.get("temphighalarm")
+                templowalarm = dom_th_info_dict.get("templowalarm")
+                temphighwarning = dom_th_info_dict.get("temphighwarning")
+                templowwarning = dom_th_info_dict.get("templowwarning")
+                if temperature != 'N/A' and temphighalarm != 'N/A' and templowalarm != 'N/A' and \
+                   temphighwarning != 'N/A' and templowwarning != 'N/A':
+                    if float(temperature) > float(temphighalarm):
+                        new_temp_status = TEMP_HIGH_ALARM
+                    elif float(temperature) > float(temphighwarning):
+                        new_temp_status = TEMP_HIGH_WARNING
+                    elif float(temperature) < float(templowalarm):
+                        new_temp_status = TEMP_LOW_ALARM
+                    elif float(temperature) < float(templowwarning):
+                        new_temp_status = TEMP_LOW_WARNING
+                    else:
+                        new_temp_status = TEMP_NORMAL
```

Comment:
@chiourung Instead of Xcvrd doing the comparison, there are module flags for low/high warning thresholds in byte 9. Why not use those latched flag?


![image](https://github.com/sonic-net/sonic-platform-daemons/assets/45705344/7b1f41ed-c65a-4f8a-8706-865447c40355)

Reply:
@prgeor These flags are COR (Clear on Read). Cannot be used to determine if the temperature is from alarm/warning to normal. The comparison of the temperature can be kept to record when the temperature is alarm / warning.